### DISCUSSION TITLE
Firmware X1

### DISCUSSION THREAD

**Discussion by HooK2002** [dgtaheno/artillery-Genius#14] [Show and tell]
Hi, great job with the Genius. Can you make the same for the X1.

**Comment by Ckarlinn** [comment]
Hola, me sumo a la petición si es posible adaptarlo para la x1, lo probe, pero al nivelar solo marca el área de la Genius.

**Reply by HooK2002** [reply]
Solo seria aumentar el area de impresion tanto en el HEX como en el config.ini de la TFT.
Lo de la TFT lo puede hacer uno mismo, pero el .HEX no se puede

**Reply by Ckarlinn** [reply]
Hola Hook, ok, si entiendo que en hex no, hay manera de mandar por pronterface el valor de la cama ? jejeje soy nuevo en esto.

**Reply by HooK2002** [reply]
El hex no se puede modificar ya ewta compilado para la genius y tendríamos que tener el fuente para modificar y compilar

**Reply by Ckarlinn** [reply]
Entiendo!!

**Reply by dgtaheno** [reply]
Hello @HooK2002 & @Ckarlinn , 

Thank you for your feedback, I have adjusted the firmware to work with the X1, if you give it a try please provide your feedback.
Please note that BLTouch is not implemented, if you need it, please let me know so I can make a version.
Here you can find the new .hex https://github.com/dgtaheno/artillery-Genius-Marlinfw/blob/Marlin-2.1.2.1/Marlin%202.1.2.1-X1.hex

Best regards and enjoy it!

**Reply by Ckarlinn** [reply]
@dgtaheno thank you very much, I will keep you informed!! thank you !!

**Reply by HooK2002** [reply]
Muchas gracias. Lo pruebo en unos minutos

**Reply by dgtaheno** [reply]
> Muchas gracias. Lo pruebo en unos minutos
> 
> 

Ya me contarás que tal funciona!

**Reply by Ckarlinn** [reply]
Me fue bien, todo en orden !! Muchas gracias... Es posible activar la recuperación de la impresión al corte de luz ? Ayer se fue la luz jejeje.

**Reply by dgtaheno** [reply]
Me alegro de que te fuese todo bien.
Una lástima que ayer se te fuese la luz.
Una solución rápida es activar la función en el slizer al iniciar la impresora mediante el parámetro M413, aquí se explica:

https://3dprinting.stackexchange.com/questions/10571/how-to-activate-power-loss-recovery-in-marlin

El único problema es que necesitas imprimir desde la SD o tener una SD conectada, para que haga los guardados.

Si trabajas con Octoprint, este plugin creo que funciona:

https://plugins.octoprint.org/plugins/powerfailure/

Un saludo.
